hack/deployer/runner/ocp.go

```diff
@@ -182,38 +182,30 @@ func (d *OcpDriver) Execute() error {
 }
 
 func (d *OcpDriver) auth() error {
+	log.Println("Authenticating as service account...")
 
-	if d.plan.ServiceAccount {
-		log.Println("Authenticating as service account...")
-
-		client, err := NewClient(*d.plan.VaultInfo)
-		if err != nil {
-			return err
-		}
+	client, err := NewClient(*d.plan.VaultInfo)
+	if err != nil {
+		return err
+	}
 
-		gcpDir := filepath.Join(os.Getenv("HOME"), ".gcp")
-		keyFileName := filepath.Join(os.Getenv("HOME"), ".gcp", "osServiceAccount.json")
-		_ = os.MkdirAll(gcpDir, os.ModePerm)
+	gcpDir := filepath.Join(os.Getenv("HOME"), ".gcp")
+	keyFileName := filepath.Join(os.Getenv("HOME"), ".gcp", "osServiceAccount.json")
+	_ = os.MkdirAll(gcpDir, os.ModePerm)
 
-		if err := client.ReadIntoFile(keyFileName, OcpVaultPath, OcpServiceAccountVaultFieldName); err != nil {
-			return err
-		}
-		log.Println("Setting $GCLOUD_KEYFILE_JSON...")
-		return os.Setenv("GCLOUD_KEYFILE_JSON", keyFileName)
+	if err := client.ReadIntoFile(keyFileName, OcpVaultPath, OcpServiceAccountVaultFieldName); err != nil {
+		return err
 	}
 
-	log.Println("Authenticating as user...")
-	accounts, err := NewCommand(`gcloud auth list "--format=value(account)"`).StdoutOnly().WithoutStreaming().Output()
-	if err != nil {
+	// ensure gcloud & gsutil rely on credentials stored in gcpDir instead of using the default
+	// directory (~/.config/gcloud), to not tamper any default gcloud auth already set on the system
+	if err := os.Setenv("CLOUDSDK_CONFIG", gcpDir); err != nil {
 		return err
 	}
-
-	if len(accounts) > 0 {
-		return nil
+	if err := NewCommand(fmt.Sprintf("gcloud config set project %s", d.ctx["GCloudProject"])).Run(); err != nil {
+		return err
 	}
-
-	_ = NewCommand(fmt.Sprintf("gcloud config set project %s", d.ctx["GCloudProject"])).Run()
-	return NewCommand("gcloud auth login").Run()
```

Comment:
Is the login action redundant? I remember it being required regardless. Did it cause any issues for you?

Reply:
`gcloud auth activate-service-account ` seems good enough when using a service account

Reply:
Hmm actually we require gcloud auth in a different place: https://github.com/elastic/cloud-on-k8s/blob/master/Makefile#L376-L378  :(
It's a bit weird to have a dependency from this Makefile to gcloud auth in the deployer.